src/app/pages/login/login.component.ts

```diff
@@ -0,0 +1,30 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { Component } from '@angular/core';
+import { Router } from '@angular/router';
+import { UserService } from 'src/app/services/user.service';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.scss'],
+})
+export class LoginComponent {
+  constructor(private userService: UserService, private router: Router) {}
+  errorAfterLogin: string = '';
+
+  async login(username: string, password: string) {
+    await this.userService
+      .login(username, password)
+      .then((res) => {
```

Comment:
Let's not use `await` and `.then()` in the same code. This can be rewritten in async style like this:
```ts
const res = await this.userService.login(username, password);
if (res) {
  this.errorAfterLogin = '';
  // ...
} else {
  // ...
}
```
The functionality for `.catch()` is provided by wrapping the whole awaited call in a `try..catch`